Tests aren't running

Steps to reproduce: 
1. download the latest version of webcompat.com
2. build the repo in whatever way you prefer (I've tried with/without virtualenv, with pyenv/pip env)
3. try to run the tests

Then the node process starts to increasingly using system's memory until it's terminated because it used all of it.

```
> webcompat@ setup /Users/mariana/Development/webcompat/tes/webcompat.com
> npm run virtualenv && npm install && npm run config


> webcompat@ virtualenv /Users/mariana/Development/webcompat/tes/webcompat.com
> pip2 install virtualenv && virtualenv -p python2.7 env && source env/bin/activate || . env/bin/activate && npm run pip

Collecting virtualenv
/Users/mariana/Development/webcompat/tes/webcompat.com/env/lib/python2.7/site-packages/pip/_vendor/urllib3/util/ssl_.py:369: SNIMissingWarning: An HTTPS request has been made, but the SNI (Server Name Indication) extension to TLS is not available on this platform. This may cause the server to present an incorrect TLS certificate, which can cause validation failures. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  SNIMissingWarning
/Users/mariana/Development/webcompat/tes/webcompat.com/env/lib/python2.7/site-packages/pip/_vendor/urllib3/util/ssl_.py:160: InsecurePlatformWarning: A true SSLContext object is not available. This prevents urllib3 from configuring SSL appropriately and may cause certain SSL connections to fail. You can upgrade to a newer version of Python to solve this. For more information, see https://urllib3.readthedocs.io/en/latest/advanced-usage.html#ssl-warnings
  InsecurePlatformWarning
  Using cached https://files.pythonhosted.org/packages/6a/d1/e0d142ce7b8a5c76adbfad01d853bca84c7c0240e35577498e20bc2ade7d/virtualenv-16.2.0-py2.py3-none-any.whl
Requirement already satisfied: setuptools>=18.0.0 in ./env/lib/python2.7/site-packages (from virtualenv) (40.6.3)
Installing collected packages: virtualenv
Successfully installed virtualenv-16.2.0
Already using interpreter /Users/mariana/Development/webcompat/tes/webcompat.com/env/bin/python2.7
Please use the *system* python to run this script
Traceback (most recent call last):
  File "/Users/mariana/Development/webcompat/tes/webcompat.com/env/bin/virtualenv", line 11, in <module>
    sys.exit(main())
  File "/Users/mariana/Development/webcompat/tes/webcompat.com/env/lib/python2.7/site-packages/virtualenv.py", line 768, in main
    symlink=options.symlink,
  File "/Users/mariana/Development/webcompat/tes/webcompat.com/env/lib/python2.7/site-packages/virtualenv.py", line 1013, in create_environment
    install_python(home_dir, lib_dir, inc_dir, bin_dir, site_packages=site_packages, clear=clear, symlink=symlink)
  File "/Users/mariana/Development/webcompat/tes/webcompat.com/env/lib/python2.7/posixpath.py", line 367, in abspath
    if not isabs(path):
  File "/Users/mariana/Development/webcompat/tes/webcompat.com/env/lib/python2.7/posixpath.py", line 61, in isabs
    return s.startswith('/')
AttributeError: 'NoneType' object has no attribute 'startswith'

> webcompat@ pip /Users/mariana/Development/webcompat/tes/webcompat.com
> pip2 install -r config/requirements.txt

Requirement already satisfied: blinker==1.4 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 1)) (1.4)
Requirement already satisfied: Flask==1.0.2 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 2)) (1.0.2)
Requirement already satisfied: Flask-Limiter==0.9.3 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 3)) (0.9.3)
Requirement already satisfied: Flask-SQLAlchemy==2.1 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 4)) (2.1)
Requirement already satisfied: Flask-WTF==0.14.2 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 5)) (0.14.2)
Requirement already satisfied: GitHub-Flask==3.1.5 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 6)) (3.1.5)
Requirement already satisfied: mock==2.0.0 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 7)) (2.0.0)
Requirement already satisfied: nose==1.3.7 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 8)) (1.3.7)
Requirement already satisfied: pep8==1.7.0 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 9)) (1.7.0)
Requirement already satisfied: Pillow==5.2.0 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 10)) (5.2.0)
Requirement already satisfied: requests==2.20.0 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 11)) (2.20.0)
Requirement already satisfied: ua-parser==0.8 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 12)) (0.8.0)
Requirement already satisfied: WTForms==2.1 in ./env/lib/python2.7/site-packages (from -r config/requirements.txt (line 13)) (2.1)
node
Tunnel: Ready

Total: Pending
Passed: 0  Failed: 0  Skipped: 0

<--- Last few GCs --->

[41283:0x105000000]   928287 ms: Mark-sweep 1314.0 (1456.8) -> 1304.6 (1456.8) MB, 1315.9 / 0.0 ms  (average mu = 0.226, current mu = 0.167) allocation failure GC in old space requested
[41283:0x105000000]   929649 ms: Mark-sweep 1304.6 (1456.8) -> 1304.5 (1456.8) MB, 1361.5 / 0.0 ms  (average mu = 0.121, current mu = 0.000) allocation failure GC in old space requested


<--- JS stacktrace --->

==== JS stack trace =========================================

    0: ExitFrame [pc: 0x1b8c4abcfb7d]
Security context: 0x280e6f108941 <JSObject>
    1: onread(aka onStreamRead) [0x280e7421a721] [internal/stream_base_commons.js:~134] [pc=0x1b8c4b1401d0](this=0x280e9d130fc9 <TCP map = 0x280e2203f2e1>,0x280e9d131e01 <ArrayBuffer map = 0x280e46c84ff1>)
    2: InternalFrame [pc: 0x1b8c4ab8ba89]
    3: EntryFrame [pc: 0x1b8c4ab84b3e]

==== Details ================================================

[0]: Exit...

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory
 1: 0x100061331 node::Abort() [/usr/local/bin/node]
 2: 0x10006194b node::OnFatalError(char const*, char const*) [/usr/local/bin/node]
 3: 0x10016926f v8::Utils::ReportOOMFailure(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 4: 0x100169210 v8::internal::V8::FatalProcessOutOfMemory(v8::internal::Isolate*, char const*, bool) [/usr/local/bin/node]
 5: 0x1004249e0 v8::internal::Heap::UpdateSurvivalStatistics(int) [/usr/local/bin/node]
 6: 0x100426421 v8::internal::Heap::CheckIneffectiveMarkCompact(unsigned long, double) [/usr/local/bin/node]
 7: 0x100423d67 v8::internal::Heap::PerformGarbageCollection(v8::internal::GarbageCollector, v8::GCCallbackFlags) [/usr/local/bin/node]
 8: 0x100422b55 v8::internal::Heap::CollectGarbage(v8::internal::AllocationSpace, v8::internal::GarbageCollectionReason, v8::GCCallbackFlags) [/usr/local/bin/node]
 9: 0x100422df5 v8::internal::Heap::CollectAllAvailableGarbage(v8::internal::GarbageCollectionReason) [/usr/local/bin/node]
10: 0x10042aa8d v8::internal::Heap::AllocateRawWithRetryOrFail(int, v8::internal::AllocationSpace, v8::internal::AllocationAlignment) [/usr/local/bin/node]
11: 0x10040b0c3 v8::internal::Factory::AllocateRawWithImmortalMap(int, v8::internal::PretenureFlag, v8::internal::Map*, v8::internal::AllocationAlignment) [/usr/local/bin/node]
12: 0x1004109ed v8::internal::Factory::NewByteArray(int, v8::internal::PretenureFlag) [/usr/local/bin/node]
13: 0x1003b2d2f v8::internal::TranslationBuffer::CreateByteArray(v8::internal::Factory*) [/usr/local/bin/node]
14: 0x10026492b v8::internal::compiler::CodeGenerator::GenerateDeoptimizationData() [/usr/local/bin/node]
15: 0x1002647c8 v8::internal::compiler::CodeGenerator::FinalizeCode() [/usr/local/bin/node]
16: 0x100325921 void v8::internal::compiler::PipelineImpl::Run<v8::internal::compiler::FinalizeCodePhase>() [/usr/local/bin/node]
17: 0x100320789 v8::internal::compiler::PipelineImpl::FinalizeCode() [/usr/local/bin/node]
18: 0x1003206c2 v8::internal::compiler::PipelineCompilationJob::FinalizeJobImpl(v8::internal::Isolate*) [/usr/local/bin/node]
19: 0x10023f4f5 v8::internal::OptimizedCompilationJob::FinalizeJob(v8::internal::Isolate*) [/usr/local/bin/node]
20: 0x1002432d6 v8::internal::Compiler::FinalizeCompilationJob(v8::internal::OptimizedCompilationJob*, v8::internal::Isolate*) [/usr/local/bin/node]
21: 0x10023d1db v8::internal::OptimizingCompileDispatcher::InstallOptimizedFunctions() [/usr/local/bin/node]
22: 0x1003f1c43 v8::internal::StackGuard::HandleInterrupts() [/usr/local/bin/node]
23: 0x1005e7eca v8::internal::Runtime_StackGuard(int, v8::internal::Object**, v8::internal::Isolate*) [/usr/local/bin/node]
24: 0x1b8c4abcfb7d
25: 0x1b8c4b1401d0
[1]    41282 abort      npm run test:js
```

If you try to run the tests again you'll get the following error:

```
> webcompat@ test:js /Users/mariana/Development/webcompat/tes/webcompat.com
> node ./tests/functional/_intern.js

something bad happened inside intern.run()
{ Error: Address is already in use
    at Socket.<anonymous> (/Users/mariana/Development/webcompat/tes/webcompat.com/node_modules/@theintern/digdug/SeleniumTunnel.js:213:28)
    at Socket.emit (events.js:193:15)
    at addChunk (_stream_readable.js:288:12)
    at readableAddChunk (_stream_readable.js:265:13)
    at Socket.Readable.push (_stream_readable.js:224:10)
    at Pipe.onStreamRead (internal/stream_base_commons.js:145:17) reported: true }
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! webcompat@ test:js: `node ./tests/functional/_intern.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the webcompat@ test:js script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/mariana/.npm/_logs/2019-01-17T16_50_36_180Z-debug.log
```

You can find the process that is using this door and close it. If you try to run the tests again you will fall into the first error.

After a few tries I found out that the problem is related to the `node_modules`. I have an old directory of webcompat.com which the last commit is 5490c9222679fb67858238d2ea844e7a7f1e03da, if I copy de `node_modules` dir from this directory to the directory containing the newer version of webcompat.com, the tests will run.

I was able to reproduce this issue in two different macs.

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -59,6 +59,7 @@ jobs:
             npm run lint
             npm run build
             node --max-old-space-size=8192 ./tests/functional/_intern.js --reporters="runner" --firefoxBinary=`which firefox`
+            firefox --version
 
       - store_artifacts:
           path: test-reports

